ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -973,11 +1090,28 @@ private void handleMetadataBlob(BlobData blobData) throws IOException, MessageFo
           numChunksTotal = 0;
           dataChunks = null;
         } else {
-          chunkIdIterator = keys.listIterator();
-          numChunksTotal = keys.size();
-          dataChunks = new GetChunk[Math.min(keys.size(), NonBlockingRouter.MAX_IN_MEM_CHUNKS)];
-          for (int i = 0; i < dataChunks.length; i++) {
-            dataChunks[i] = new GetChunk(chunkIdIterator.nextIndex(), (BlobId) chunkIdIterator.next());
+          // if blob is encrypted, then decryption is required only in case of GetBlobInfo and GetBlobAll (since user-metadata
+          // is expected to be encrypted). Incase of GetBlob, Metadata blob does not need any decryption even if BlobProperties says so
```

Comment:
I wonder if things are simple if we just encrypted metadata blob as well (small price to pay for code simplicity)

Reply:
yes, I thought the same too. During our design discussions, someone raised a point why do we encrypt Metadata chunk if it doesn't contain PII and hence we decided not go encrypt Metadata blobs. and hence I am following the same for now. I am okay in encrypting both content and userMetadata for Metadata chunk